modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,137 @@
+Each different hardware platform requires a separate seed image.
+====
+// TODO agree on how much we should specify this. Is this enough or detail about CPU topology, deployment method, etc.
+
+The {lcao} uses two custom resources (CRs) on the participating clusters to orchestrate the upgrade:
+
+* On the seed cluster, the `SeedGenerator` CR allows for the seed image generation. This CR specifies the repository to push the seed image to.
+* On the target cluster, the `ImageBasedUpgrade` CR specifies the seed container image for the upgrade of the target cluster and the backup configurations for your workloads.
+
+.Example SeedGenerator CR
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage
+spec:
+  seedImage: <seed_container_image>
+----
+
+.Example ImageBasedUpgrade CR
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+spec:
+  stage: Idle <1>
+  seedImageRef: <2>
+    version: <target_version>
+    image: <seed_container_image>
+    pullSecretRef: <seed_pull_secret>
+  additionalImages:
+    name: ""
+    namespace: ""
+  autoRollbackOnFailure: {} <3>
+#    disabledForPostRebootConfig: "true" <4>
+#    disabledForUpgradeCompletion: "true" <5>
+#    disabledInitMonitor: "true" <6>
+#    initMonitorTimeoutSeconds: 1800 <7>
+#  extraManifests: <8>
+#  - name: sno-extra-manifests
+#    namespace: openshift-lifecycle-agent
+#  oadpContent: <9>
+#  - name: oadp-cm-example
+#    namespace: openshift-adp
+----
+<1> Defines the desired stage for the `ImageBasedUpgrade` CR. The value can be `Idle`, `Prep`, `Upgrade`, or `Rollback`.
+<2> Defines the target platform version, the seed image to be used, and the secret required to access the image.
+<3> Configures the automatic rollback. By default, automatic rollback on failure is enabled throughout the upgrade.
+<4> (Optional) Disables automatic rollback when the reconfiguration of the cluster fails upon the first reboot.
+<5> (Optional) Disables automatic rollback after the Lifecycle Agent Operator reports a failed upgrade upon completion.
+<6> (Optional) Disables automatic rollback when the upgrade does not complete after reboot within the time frame specified in the `initMonitorTimeoutSeconds` field.
+<7> (Optional) Specifies the time frame in seconds. If not defined, the default value is 1800 seconds or 30 minutes.
+<8> (Optional) Specify the list of `ConfigMap` CRs that contain the additional extra manifests that you want to apply to the target cluster.
+<9>	(Optional) Specify the list of `ConfigMap` CRs that contain the OADP `Backup` and `Restore` CRs.
+
+After generating the seed image on the seed cluster, you can move through the stages on the target cluster by specifying the image to pull and then setting the `spec.stage` field to the following values in the `ImageBasedUpgrade` CR:
```

Comment:
It's not possible to move between some stages.
For example:
   can't move from Upgrade to Prep.
   can't move from Rollback to Prep.

Do we want to mention it.